### PR TITLE
overlays: add wm8960-soundcard overlay

### PR DESCRIPTION
**Pull request by shawaj**
add overlay for waveshare wm8960 simple-audio-card

Change-type: patch
Signed-off-by: Aaron Shaw <shawaj@gmail.com>

### CODE REVIEW

**Comment by shawaj** [comment]
FYI - in the readme I left the description for the `master` override as unsure of function - because I wasn't sure what it actually does. 

Additionally, this overlay will also bring support for (they are all clones):
- [Seeed 2 mic voicecard](https://github.com/respeaker/seeed-voicecard/blob/master/seeed-2mic-voicecard-overlay.dts)
- [keyestudio respeaker 2 mic](https://wiki.keyestudio.com/index.php/Ks0314_keyestudio_ReSpeaker_2-Mic_Pi_HAT_V1.0)
- [Waveshare WM8960](https://github.com/waveshare/WM8960-Audio-HAT)

File: arch/arm/boot/dts/overlays/README [hunk 1]
```diff
@@ -3033,6 +3033,14 @@ Params: int_pin                 GPIO used for INT (default 25)
         cs                      SPI bus Chip Select (default 0)
 
 
+Name:   wm8960-soundcard
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README]: Alphabetical order, please.

File: arch/arm/boot/dts/overlays/Makefile [hunk 2]
```diff
@@ -209,6 +209,7 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	w1-gpio.dtbo \
 	w1-gpio-pullup.dtbo \
 	w5500.dtbo \
+	wm8960-soundcard.dtbo \
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/Makefile]: "wm" comes after "wi".

File: arch/arm/boot/dts/overlays/wm8960-soundcard-overlay.dts [hunk 3]
```diff
@@ -0,0 +1,84 @@
+// Definitions for Waveshare WM8960 https://github.com/waveshare/WM8960-Audio-HAT
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "brcm,bcm2708";
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/wm8960-soundcard-overlay.dts]: Please use `"brcm,bcm2835"`.

File: arch/arm/boot/dts/overlays/wm8960-soundcard-overlay.dts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+	fragment@1 {
+		target-path="/";
+		__overlay__ {
+			wm8960_mclk: wm8960_mclk {
+				compatible = "fixed-clock";
+				#clock-cells = <0>;
+				clock-frequency = <12288000>;
+			};
+
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/wm8960-soundcard-overlay.dts]: Delete this blank line.

File: arch/arm/boot/dts/overlays/wm8960-soundcard-overlay.dts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+		};
+	};
+	fragment@2 {
+		target = <&i2c1>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "okay";
+
+			wm8960: wm8960{
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/wm8960-soundcard-overlay.dts]: There should be a space before the brace.

**Review by pelwell — CHANGES_REQUESTED** [review]
You need to understand and explain what the "master" parameter is doing - it looks wrong to me because if set to "on" it enables fragment 2 and disables fragment 3, and if "off" it disables fragment 2 and enables fragment 3, and yet both fragments are enabled by default.

File: arch/arm/boot/dts/overlays/Makefile (hunk 2 above)
Comment (shawaj) [reply arch/arm/boot/dts/overlays/Makefile]: doh! my bad 🤦‍♂️

**Comment by shawaj** [comment]
@pelwell have updated the PR. As far as I can tell - the master override is not needed. I can't find anything in the docs for the board about using it or anything like that so I also agree that it is wrong, certainly for this board. Perhaps it is used for some other hardware that I am not aware of, but I can't see any function for it either.

**Comment by pelwell** [comment]
You can always add it later if a use becomes apparent.

**Comment by shawaj** [comment]
agreed, thanks @pelwell

**Comment by tewarid** [comment]
Thanks for adding this. It works with Seeed's [ReSpeaker 2-Mics Pi HAT](http://wiki.seeedstudio.com/ReSpeaker_2_Mics_Pi_HAT/) after enabling a few switches that are off, and adjusting the volume

```bash
amixer -c1 sset 'Left Input Mixer Boost' toggle
amixer -c1 sset 'Left Output Mixer PCM' toggle
amixer -c1 sset 'Right Input Mixer Boost' toggle
amixer -c1 sset 'Right Output Mixer PCM' toggle
amixer -c1 sset 'Headphone',0 100%,100%
```

I am, however, getting very noisy sound capture using

```bash
arecord -f cd -Dhw:1 -d5 > test.wav
```

Please see [test.wav.zip](https://github.com/raspberrypi/linux/files/6257795/test.wav.zip).

I don't get that when using the [Seeed's official soundcard driver](https://github.com/respeaker/seeed-voicecard). I've compared alsa settings and they look all right to my, untrained, eyes. See  [asound.state.txt](https://github.com/raspberrypi/linux/files/6257874/asound.state.txt) and [seeed-asound.state.txt](https://github.com/raspberrypi/linux/files/6257875/seeed-asound.state.txt).

**Comment by meiser79** [comment]
Hi,

this overlay needs an update, see https://github.com/jxlarrea/kiosk-satellite/issues/563#issuecomment-5750401495

Regards,
meiser79

**Pull request merged**